src/nodejs/hooks/seeker_agent_test.go

```diff
@@ -0,0 +1,479 @@
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				Expect(ioutil.WriteFile(entryPointPath, []byte("some mock javascript code"), 0755)).To(Succeed())
+				err = seeker.AfterCompile(stager)
+				Expect(err).To(BeNil())
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(
+					"require('./seeker/node_modules/@synopsys-sig/seeker-inline');\n" +
+						"some mock javascript code\n"))
+			})
+			It("Fails to prepend the require to the server.js file - when the file does not exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				err = seeker.AfterCompile(stager)
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).To(ContainSubstring("no such file or directory"))
+			})
+
+		})
+	})
+})
+
+func assertSensorDownload() {
+	Expect(httpmock.GetCallCountInfo()["GET "+getVersionURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()["GET "+getSensorURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()).ShouldNot(ContainElement("GET " + getAgentURL()))
+	Expect(httpmock.GetTotalCallCount()).To(Equal(2))
+}
+func assertAgentDownload() {
+	Expect(httpmock.GetCallCountInfo()["GET "+getVersionURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()["GET "+getAgentURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()).ShouldNot(ContainElement("GET " + getSensorURL()))
+	Expect(httpmock.GetTotalCallCount()).To(Equal(2))
+}
+
+func getAgentZip() ([]byte, error) {
+	return getFixtureContent("NODEJS_agent.zip")
+}
+func getSensorZip() ([]byte, error) {
+	return getFixtureContent("NODEJS_SensorWithAgent.zip")
+}
+func getFixtureContent(fileName string) ([]byte, error) {
+	path, err := filepath.Abs("../../../fixtures/seeker/" + fileName)
+	if err != nil {
+		return nil, err
+	}
+	fileContent, err := ioutil.ReadFile(path)
+	return fileContent, err
+}
+
+func mockSeekerVersionThatSupportsOnlySensorDownload() {
+	httpmock.RegisterResponder("GET", getVersionURL(),
+		httpmock.NewStringResponder(200, `{"publicName":"Seeker Enterprise Server","version":"2018.05-SP1-SNAPSHOT","buildNumber":"20180629131550","scmBranch":"origin/release/v2018.06","scmRevision":"815ba309"}`))
+}
+
+func mockSeekerVersionThatSupportsAgentDownload() {
+	httpmock.RegisterResponder("GET", getVersionURL(),
+		httpmock.NewStringResponder(200, `{"publicName":"Seeker Enterprise Server","version":"2018.06-SNAPSHOT","buildNumber":"20180629131550","scmBranch":"origin/release/v2018.06","scmRevision":"815ba309"}`))
+}
+func mockSensorDownload() {
+	zipContent, _ := getSensorZip()
+	httpmock.RegisterResponder("GET", getSensorURL(),
+		httpmock.NewBytesResponder(200, zipContent))
+}
+
+func mockAgentDownload() {
+	zipContent, _ := getAgentZip()
+	httpmock.RegisterResponder("GET", getAgentURL(),
+		httpmock.NewBytesResponder(200, zipContent))
+}
```

Comment:
Generally, we would prefer not to use httpmock. We would encourage you to make a `Downloader` interface for the `SeekerAfterCompileHook` object that you can mock in the unit tests. 

Ex

```
type DownloaderThing interface {
	DownloadFile(url, path string) error
}

type SeekerAfterCompileHook struct {
	libbuildpack.DefaultHook
	Log                *libbuildpack.Logger
	serviceCredentials *SeekerCredentials
	Command            SeekerCommand
	Downloader	   *DownloaderThing
}
```

Unit tests could just set the `Downloader` object on `SeekerAfterCompileHook` to a mock object.

Reply:
fixed